Lesson15/duygu/script.js

```diff
@@ -0,0 +1,631 @@
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+function censorWord(sentence, censored){
+  console.log (sentence.replaceAll(censored, '****'));
+}
+censorWord("This is a censored word.", "censored");
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+function startsWithA(str) {
+  if (str.charAt(0) === 'A') {
+  console.log("Starts with A");
+}
+ else {
+  console.log("Does not start with A");
+}
+}
+startsWithA("Ayakkabı");
+startsWithA("Pantolon");
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+function sliceLastN(text, n) {
+  const nthword = text.slice(n)
+  console.log(nthword);
+}
+sliceLastN("How are you?", -4);
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+function gradeChecker(score) {
+  if (score >= 90) {
+    console.log("A");
+  } else if (score >= 80) {
+    console.log("B");
+  } else if (score >= 70) {
+    console.log("C");
+  } else if (score >= 60) {
+    console.log("D");
+  } else {
+    console.log("F");
+  }
+}
+
+gradeChecker(95);
+gradeChecker(85);
+gradeChecker(75);
+gradeChecker(65);
+gradeChecker(50);
+
+/*
+18. Character Replacement
+   - Define a function `replaceCharacter(str, oldChar, newChar)` that uses .replaceAll()
+     (or a loop) to swap all occurrences of oldChar with newChar.
+   - Log the result.
+*/
+function replaceCharacter(word, oldChar, newChar){
+  console.log (word.replaceAll(oldChar, newChar));
+  
+}
+replaceCharacter("Anlamadım", "n", "ğ");
+/*
+19. Title Case a Sentence
+   - Define a function `titleCase(sentence)` that:
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word
+     - Joins them back
+   - Log the transformed string.
+*/
+function titleCase(sentence) {
+  let words = sentence.split(" ");
+  let capitalizedWords = words.map(word => 
+    word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
+  );
+  let transformedSentence = capitalizedWords.join(" ");
+
+  console.log(transformedSentence);
+  return transformedSentence;
+}
+
+
+titleCase("hello world");
+titleCase("this is a test");
+titleCase("javaScript IS fun!");
+
+
+/*
+20. Switch: Traffic Light
+   - Define a function `trafficLight(color)` that uses a switch statement:
+     - "red" -> log: "Stop"
+     - "yellow" -> log: "Caution"
+     - "green" -> log: "Go"
+     - anything else -> "Invalid color"
+*/
+function trafficLight(color) {
+  switch (color) {
```

Comment:
It would be better to make this function less error-prone by converting the incoming color value to lowercase. Then, it would still work for this test: `trafficLight("BLUE");`
```suggestion
  switch (color.toLowerCase()) {
```